pkg/daemon/daemon.go

```diff
@@ -507,18 +507,22 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 				go dn.getDrainLock(ctx, done)
 				<-done
 			}
+		}
 
-			if dn.openshiftContext.IsOpenshiftCluster() && !dn.openshiftContext.IsHypershift() {
-				glog.Infof("nodeStateSyncHandler(): pause MCP")
-				if err := dn.pauseMCP(); err != nil {
-					return err
-				}
+		if dn.openshiftContext.IsOpenshiftCluster() && !dn.openshiftContext.IsHypershift() && !dn.isMcpPaused() {
```

Comment:
why do we need this change ? looking at pauseMCP it should handle an already paused MCP object (which is paused by  config daemon based on node annot)


btw, in MCPPause if we manage to pause MCP but fail to update annotation i think we will deadlock.
here: https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/109c2c25df7ac335bfd7a76ef3ee43ef69c95c82/pkg/daemon/daemon.go#L843

Reply:
regarding PauseMCP doing early return in case node drain annot is mcp_paused: 
https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/109c2c25df7ac335bfd7a76ef3ee43ef69c95c82/pkg/daemon/daemon.go#L839

Reply:
here is what i think happens

node-1 : 
1. get lock
2. annotate node as draning
3. try to pause mcp
4. reboot (it did not pause mcp), hence it did not set Draining_MCP_Paused

node-2:
1. tries to get lock <- stuck as there is not draning, but it holds the lock

node-1:
1. config daemon comes back up
2. isNodeDraining returns true
3. calls PauseMCP()
4. PauseMCP() now completes successfully
5. config daemon continues with configuration
6. eventually completes drain, unpause MCP and set node draining annot to Idle

node-2:
1. continues its work


what did i get wrong ?


where i think a deadlock may happen:

node-1 : 
1. get lock
2. annotate node as draning
3. try to pause mcp
4. patch mcp object setting pause to true
5. reboot (it did not set Draining_MCP_Paused node annot)

node-2:
1. tries to get lock <- stuck as there is not draning, but it holds the lock

node-1:
1. config daemon comes back up
2. isNodeDraining returns true
3. calls PauseMCP()
4. stuck as we thing someone else paused MCP obj